### ISSUE TITLE
Budget Undefined Categories

### ISSUE THREAD

**Issue by drunnells** [mrooney/mintapi#96]
Thank you for this great script! I've been using it successfully and even integrated it into an [Alexa Skill](https://github.com/drunnells/alexa-mint-skill). However, budgets don't seem to be working quite right for me. mintapi fetches my budgets, but the returned JSON leaves *most* of the cats as "Undefined". All of my budgets seem to be here and have the correct amounts, but I can't determine which budget they belong to.

## Command  ##
```shell
# /opt/python-2.7.2/bin/mintapi --budgets --session 123412341234 --thx_guid 123412341234 user@example.com password
```

## Result ##
```json
{
  "spend": [
    {
      "catTypeFilter": "Personal", 
      "ramt": 0, 
      "cat": "Gas & Fuel", 
      "pid": 14, 
      "amt": 82.16, 
      "isIncome": false, 
      "isTransfer": false, 
      "bgt": 140.0, 
      "ex": false, 
      "id": 56418315, 
      "st": 1, 
      "type": 0, 
      "isExpense": true, 
      "rbal": 57.84
    }, 
    {
      "catTypeFilter": "Personal", 
      "ramt": 0, 
      "cat": "Unknown", 
      "pid": null, 
      "amt": 90, 
      "isIncome": false, 
      "isTransfer": false, 
      "bgt": 700.0, 
      "ex": false, 
      "id": 56418316, 
      "st": 1, 
      "type": 0, 
      "isExpense": true, 
      "rbal": 610.0
    }, 
    {
      "catTypeFilter": "Personal", 
      "ramt": 0, 
      "cat": "Unknown", 
      "pid": null, 
      "amt": 96.25, 
      "isIncome": false, 
      "isTransfer": false, 
      "bgt": 150.0, 
      "ex": false, 
      "id": 56418312, 
      "st": 1, 
      "type": 0, 
      "isExpense": true, 
      "rbal": 53.75
    }, 
    {
      "catTypeFilter": "Personal", 
      "ramt": 0, 
      "cat": "Unknown", 
      "pid": null, 
      "amt": 0, 
      "isIncome": false, 
      "isTransfer": false, 
      "bgt": 50.0, 
      "ex": false, 
      "id": 56418311, 
      "st": 1, 
      "type": 0, 
      "isExpense": true, 
      "rbal": 50.0
    }
  ]
}

**Comment by drunnells** [comment]
Looking a little closer, it seems like the categories that I'm missing are being intentionally avoided by get_categories() on line 515 and 516 where it checks if the parentId is 0 and then continues:

```python
        # Build category list
        categories = {}
        for category in response['allCategories']:
            if category['parentId'] == 0:
                continue
            categories[category['id']] = category
```

If I comment out those two lines I get the results that I was expecting. But I don't know if it would be dangerous to do so? 

```diff
--- api.py.orig 2017-04-14 15:46:38.000000000 -0400
+++ api.py      2017-04-14 16:20:48.000000000 -0400
@@ -512,8 +512,8 @@
         # Build category list
         categories = {}
         for category in response['allCategories']:
-            if category['parentId'] == 0:
-                continue
+            #if category['parentId'] == 0:
+            #    continue
             categories[category['id']] = category
 
         return categories
```

**Comment by mrooney** [comment]
I'm not too familiar with that code, but if it fixes that issue, I'm happy to remove it. If it had an important reason, the original author should have added a comment :) If it causes another bug, we'll address it when reported.

Merged and released as 1.24, thanks!
